Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/Processes/Validation/Down.mo

```diff
@@ -0,0 +1,1002 @@
+
+  Buildings.Controls.OBC.CDL.Logical.TrueFalseHold yStaChaPro1(
+    final trueHoldDuration=10,
+    final falseHoldDuration=0)
+    "Hold true pulse signal for visualization"
+    annotation (Placement(transformation(extent={{250,190},{270,210}})));
+
+  Buildings.Controls.OBC.CDL.Logical.TrueFalseHold yStaChaPro4(
+    final trueHoldDuration=10,
+    final falseHoldDuration=0)
+    "Hold true pulse signal for visualization"
+    annotation (Placement(transformation(extent={{-120,-146},{-100,-126}})));
+
+protected
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant uBoi[nBoi](
+    final k={true,true})
+    "Initial boiler status"
+    annotation (Placement(transformation(extent={{-260,170},{-240,190}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant con1[nBoi](
+    final k={false,true})
+    "Boiler setpoints for stage 2"
+    annotation (Placement(transformation(extent={{-334,70},{-314,90}})));
+
+  Buildings.Controls.OBC.CDL.Logical.LogicalSwitch logSwi[nBoi]
+    "Pass initial boiler status and switch to signal from controller"
+    annotation (Placement(transformation(extent={{-220,190},{-200,210}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat[nBoi](
+    final pre_y_start=fill(false, nBoi))
+    "Hold true signal after change in boiler status"
+    annotation (Placement(transformation(extent={{-290,190},{-270,210}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Pre pre3[nBoi]
+    "Logical pre block"
+    annotation (Placement(transformation(extent={{-260,220},{-240,240}})));
+
+  Buildings.Controls.OBC.CDL.Logical.LogicalSwitch logSwi1[nBoi]
+    "Pass boiler setpoints for stage 3 after one stage change"
+    annotation (Placement(transformation(extent={{-220,100},{-200,120}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant con2[nBoi](
+    final k={true,false})
+    "Boiler setpoints for stage 3"
+    annotation (Placement(transformation(extent={{-260,120},{-240,140}})));
+
+  Buildings.Controls.OBC.CDL.Logical.FallingEdge falEdg
+    "Falling edge detector"
+    annotation (Placement(transformation(extent={{-90,150},{-70,170}})));
+
+  Buildings.Controls.OBC.CDL.Logical.IntegerSwitch intSwi
+    "Switch between stage 2 and stage 3 setpoints"
+    annotation (Placement(transformation(extent={{-220,30},{-200,50}})));
+
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt(
+    final k=1)
+    "Stage 3 setpoint"
+    annotation (Placement(transformation(extent={{-260,50},{-240,70}})));
+
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt1(
+    final k=2)
+    "Stage 2 setpoint"
+    annotation (Placement(transformation(extent={{-368,0},{-348,20}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Or or2
+    "Detect first triggering pulse and subsequent stage change completion signals"
+    annotation (Placement(transformation(extent={{-130,110},{-110,130}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Switch swi[nBoi]
+    "Pass initial valve position and switch to signal from controller"
```

Comment:
this is a bit unclear just by looking at the text, I leave it up to you to edit as needed.